src/__tests__/integration/upload.test.js

```diff
@@ -0,0 +1,76 @@
+import { integrationTest, waitFor, uploadFile } from './index';
+import { act } from 'react-dom/test-utils';
+
+const EMPTY_DROPZONE_TEXT = 'Drop some PDF files here, or click to select files';
+const TEST_FILE = {
+    path: './src/__tests__/integration/assets/test.pdf',
+    name: 'test.pdf',
+    type: 'application/pdf',
+    size: '30.56 KB',
+};
+const FAILED_FILE = {
+    ...TEST_FILE,
+    name: '',
+};
+
+const isFileUploaded = state => state.pdfFiles.length;
+const isStoredInDB = store => store.getState().pdfFiles[0].isStored;
+const isBeforeUnloadAppeared = () => {
```

Comment:
This utility is not a simple getter, it "does" some stuff. So, let's rename it to `simulateUnload`.